### PR TITLE
👷 Feat: Replace Travis w/ GitHub Actions CI

### PR DESCRIPTION
**Pull request by Jabster28**
For Synk to work, you may need to add `SYNK_TOKEN` as one of your secrets in the repo settings

Also, you might want to squash merge this instead of normal merging since there's a lot of commits with not-so-descriptive names (i was tired :P)

### CODE REVIEW

**Comment by GitSquared** [comment]
I think we're going to scrap Snyk altogether, its functionality has been mostly replaced when GitHub bought Dependabot and integrated CVE warnings in its core product...

File: .github/workflows/darwin.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    runs-on: macos-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps:
+    - uses: actions/checkout@v1 # Version range or exact version of a Python version to use, using SemVer's version range syntax
+    - uses: actions/setup-python@v2
```
Comment (GitSquared) [inline .github/workflows/darwin.yaml:19]: Do we really need Python on all platforms?

**Comment by GitSquared** [comment]
Wishlist:
- [ ] Cache node_modules and Electron binaries, possibly even minified source files?
- [ ] Workflow for automatically uploading artifacts on tagged release
- [ ] Delete Travis!

Thanks for contributing this, i'm going to see if I can improve the points above and then merge. :heart:

File: .github/workflows/darwin.yaml (hunk 1 above)
Comment (Jabster28) [reply .github/workflows/darwin.yaml:19]: Not sure, I think we need it on Linux tho - I'll do some testing later

**Comment by Jabster28** [comment]
Would you mind if I combine all of the workflow files into one and used `matrix` instead? It would make uploading artefacts a lot easier

File: .github/workflows/darwin.yaml (hunk 1 above)
Comment (Jabster28) [reply .github/workflows/darwin.yaml]: Huh, it doesn't seem like it's needed on any platforms - all of my builds on [this commit](https://github.com/Jabster28/edex-ui/commit/47da54be1a6e808051f01663eed8e282463eacf7) worked without it.

**Comment by Jabster28** [comment]
Correct me if I'm wrong, but it seems like [this gh action](https://github.com/marketplace/actions/upload-files-to-a-github-release) makes a release and immediately publishes it, and it won't allow for multiple file uploads (i.e the only binaries in the release will be the first one to finish building). Is there any way we could use something built-in, like [this?](https://www.electron.build/configuration/publish)

**Comment by GitSquared** [comment]
Yeah, we could leverage electron-builder for this. Can we use the github token auto-generated for Actions or do I need to setup a custom one..?

**Comment by Jabster28** [comment]
I'm fairly sure that the auto-generated one will work, I'll do some tests in a minute and let you know

File: .github/workflows/darwin.yaml (hunk 1 above)
Comment (GitSquared) [reply .github/workflows/darwin.yaml]: Well great news then! One less thing that can break ;)

**Comment by Jabster28** [comment]
I'm getting a 403, which is (hopefully?) because I don't own the repository I forked (GitHub is weird with permissions on GH Action forks - see [here](https://github.com/actions/first-interaction/issues/10)), so it should work once merged since i'm using the default github token.

**Comment by Jabster28** [comment]
Screenshot 2020-11-26 at 08 41 56
Should we remove all of these?

**Comment by GitSquared** [comment]
There's 370 mentions of snyk?? where?

**Review by GitSquared — APPROVED** [review]
Unless you want to add something, I'm ready to merge this!

**Comment by Jabster28** [comment]
Nope, looks good to me!

**Comment by GitSquared** [comment]
Thanks a lot for helping out with this!

**Pull request merged**